utils/socket.h

```diff
@@ -0,0 +1,14 @@
+#ifndef UFTRACE_SOCKET_H
+#define UFTRACE_SOCKET_H
+
+#include <sys/socket.h>
+
+#include "uftrace.h"
```

Comment:
Please don't include this unnecesarily.  If it's only for `struct uftrace_opts`, we can have a forward declaration.